extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -313,4 +315,13 @@ HealthBuildItem addHealthCheck(MongoClientBuildTimeConfig buildTimeConfig) {
         return new HealthBuildItem("io.quarkus.mongodb.health.MongoHealthCheck",
                 buildTimeConfig.healthEnabled, "mongodb");
     }
+
+    @BuildStep
+    void produceMetricsEnabledProperty(BuildProducer<SystemPropertyBuildItem> producer,
+            MongoClientBuildTimeConfig buildTimeConfig, Capabilities capabilities) {
```

Comment:
AFAIK the full list of metrics is known at build time? If so, we should register them here (including their tags). See https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java#L427 for example

Reply:
> AFAIK the full list of metrics is known at build time?
We do not necessarily know all tags.
We use the tags host and port here.

In case of the mongo+srv protocoll, the servers are resolved through dns entries.
https://docs.mongodb.com/manual/reference/connection-string/#dns-seedlist-connection-format
We therefore would not know them during build time.

Reply:
Ok, that's unfortunate. In that case, the metrics won't currently get automatically deregistered during reload in dev mode (until we include a commit for https://github.com/quarkusio/quarkus/issues/7018), therefore `MongoMetricsConnectionPoolListener` will at least temporarily need to take into account the fact that after a reload, the metrics from previous run can still exist, and instead of just registering them, it should delete and re-register them if they exist, so that they can start again from zero